sql/table.cc

```diff
@@ -8522,7 +8522,14 @@ void TR_table::store_data(ulonglong trx_id, ulonglong commit_id, timeval &commit
   store(FLD_COMMIT_ID, commit_id);
   store(FLD_COMMIT_TS, commit_ts);
   store(FLD_BEGIN_TS, start_time);
-  store(FLD_ISO_LEVEL, thd->tx_isolation);
+  store_iso_level(thd->tx_isolation);
+}
+
+enum_tx_isolation TR_table::iso_level() const
+{
+  enum_tx_isolation res= (enum_tx_isolation) ((*this)[FLD_ISO_LEVEL]->val_int() - 1);
```

Comment:
Looks like the order of enum values in table definition matters. Maybe it's worth commenting that in table definition?

Reply:
Please, update schema check to verify order of enum values.